cli-plugins/manager/hooks.go

```diff
@@ -81,18 +89,37 @@ func invokeAndCollectHooks(dockerCli command.Cli, rootCmd, subCmd *cobra.Command
 	return nextSteps
 }
 
-func registersHook(pluginCfg map[string]string, subCmdName string) bool {
-	hookCmdStr, ok := pluginCfg["hooks"]
+func parsePluginCommandStr(plugin string, args []string) (ret string) {
+	ret = plugin
+	if len(args) < 2 {
+		return
+	}
+	for _, arg := range args[1:] {
+		ret += " " + arg
+	}
+	return ret
+}
+
+// pluginMatch takes a plugin configuration and a string representing the
+// canonical command being executed (such as 'image ls' – the root 'docker'
+// is omitted) and, if the configuration includes a hook for the invoked
+// command, returns the configured hook string,
+// In the case that the command being executed is a plugin, the command
+// string should being with the plugin name, such as 'compose' or 'buildx'.
+func pluginMatch(pluginCfg map[string]string, subCmd string) (string, bool) {
+	configuredPluginHooks, ok := pluginCfg["hooks"]
 	if !ok {
-		return false
+		return "", false
 	}
-	commands := strings.Split(hookCmdStr, ",")
+
```

Comment:
Providing an empty value here (`"hooks": ""}` will execute the hooks for **all** docker commands.

Perhaps we should also just ignore it in that case?